### PR TITLE
Format `DictComp` expression

### PR DESCRIPTION
**Pull request by cnpryer**
## Summary

Format `DictComp` like `ListComp` from #5600. It's not 100%, but I figured maybe it's worth starting to explore.

## Test Plan

Added ruff fixture based on `ListComp`'s.

### CODE REVIEW

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap [hunk 1]
```diff
@@ -331,24 +331,19 @@ last_call()
...
--}
-+{NOT_IMPLEMENTED_dict_key: NOT_IMPLEMENTED_dict_value for key, value in NOT_IMPLEMENTED_dict}
-+{NOT_IMPLEMENTED_dict_key: NOT_IMPLEMENTED_dict_value for key, value in NOT_IMPLEMENTED_dict}
-+{NOT_IMPLEMENTED_dict_key: NOT_IMPLEMENTED_dict_value for key, value in NOT_IMPLEMENTED_dict}
-+{NOT_IMPLEMENTED_dict_key: NOT_IMPLEMENTED_dict_value for key, value in NOT_IMPLEMENTED_dict}
-+{NOT_IMPLEMENTED_dict_key: NOT_IMPLEMENTED_dict_value for key, value in NOT_IMPLEMENTED_dict}
++    for (
++        k,
++        v,
++    ) in this_is_a_very_long_variable_which_will_cause_a_trailing_comma_which_breaks_the_comprehension
```
Comment (cnpryer) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap:342]: Note the generators formatting.

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap [hunk 2]
```diff
@@ -607,11 +596,17 @@ str or None if (1 if True else 2) else str or bytes or None
...
+{i: 0 for i in (1, 2, 3)}
+{i: j for i, j in ((1, "a"), (2, "b"), (3, "c"))}
+{a: b * 2 for a, b in dictionary.items()}
+{a: b * -2 for a, b in dictionary.items()}
+{
+    k: v
+    for (
+        k,
+        v,
+    ) in this_is_a_very_long_variable_which_will_cause_a_trailing_comma_which_breaks_the_comprehension
```
Comment (cnpryer) [inline crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap:608]: Note the generators formatting.

File: crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap (hunk 1 above)
Comment (cnpryer) [reply crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap:342]: cc: @davidszotten

Comment (MichaReiser) [reply crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__expression.py.snap:342]: You mean the parentheses around the tuple and that it expands over multiple lines? We probably need to use the `StripInsideForLoop` layout here as well (unrelated to your PR)

https://github.com/astral-sh/ruff/issues/5779

**Review by MichaReiser — APPROVED** [review]
Nice! Thank you.

**Pull request merged**